### ISSUE TITLE
Black screen instead of video after reconnect

### ISSUE THREAD

**Issue by sergeylaptev** [blackuy/react-native-twilio-video-webrtc#182]
### Steps to reproduce
1. Connect to the room
2. Disconnect
3. Connect to any other room

### Expected behaviour
All participants and local video should be visible

### Actual behaviour
All participants whose have enabled video track (including myself) have black screens. The issue exists only on Android, iOS works fine. I've noticed, that if I'll postpone connection for about 5 seconds after disconnect, everything works fine. Is there sort of async process in the native android module?

### Environment
- **Node.js version**:
- **React Native version**:
- **React Native platform + platform version**: RN 0.57.8, Android 8

### react-native-twilio-video-webrtc
**Version**: master

**Comment by sergeylaptev** [comment]
More info about the issue:
If you will wait about 5 seconds after disconnect and then connect to any room, you will see other participants video tracks as expected, but YOUR local video will stay black. To fix this I'm also unmounting the TwilioVideo Component, and mount it again before subsequent connect call.

**Comment by sergeylaptev** [comment]
Android seems to have race conditions while unmounting. Would be great to have callback on disconnect handler at least. @slycoder do you have any ideas why does this work exactly like this?

Below I attach the code example of my temporary workaround:
```js
class TwilioVideoComponent extends Component {
  state = {
    readyToConnect: true
  }

  componentWillUnmount () {
    this.disconnect()
  }

  connect (data) {
    this.twilioRef.connect(data)
  }

  reconnect (data) {
    if (!this.state.readyToConnect) return

    this.disconnect()

    this.setState({ readyToConnect: false })

    setTimeout(() => {
      if (!this.__mounted) return

      this.setState({ readyToConnect: true }, () => {
        this.connect(data)
      })
    }, 3000)
  }

  disconnect () {
    this.twilioRef.disconnect()
  }

  handleTwilioRef = (ref) => {
    this.twilioRef = ref
  }

  render () {
    const { readyToConnect } = this.state

    if (!readyToConnect) {
      return null
    }

    return <TwilioVideoRN {...this.props.props} ref={this.handleTwilioRef} />
  }
}
```

**Comment by harrydema** [comment]
Actualy you have the callback, 

```
<TwilioVideoRN
onRoomDidDisconnect={(foo) => {
     console.log('bar');
}}
{...this.props.props}
ref={this.handleTwilioRef} 
/>
```

**Comment by jedabero** [comment]
I seem to be having this issue too

@harrydema the callback does get fired, but when you try to connect after connecting the user camera appears black

**Comment by jedabero** [comment]
To anyone stumbling into this issue, in my case the TwilioVideo is unmounting too fast before the disconnect process finishes (I'm calling `this.props.navigation.goBack()`)
A solution for that situation is to wait around 5 seconds before calling any method that causes the unmounting, those 5 seconds are just a guess, it could be longer, maybe that depends on the network speed?

**Comment by jedabero** [comment]
I tried making a simple repo showing the issue with the fix commented: https://github.com/pluriza/twilio-view-shot
Use it to see the problem when connecting again
To see it fixed, just uncomment the `disconnected` state and the lines in startCall and closeCall

**Comment by jedabero** [comment]
@blackuy @slycoder any thoughts?

**Comment by slycoder** [comment]
Just to check, you don't try the second connection until after onRoomDidDisconnect is fired right? Afaik, that's the only signal twilio gives us that whatever it needs to do is complete.  If that doesn't work then it must be deeper in twilio in a way that's not exposed to us.

**Comment by jedabero** [comment]
@slycoder I'll check in the repo I made so the users re-connect only after onRoomDidDisconnect is fired.
In the end in the other project the issue on reconnect was that I was using react-native-camera to show a camera preview before actually connecting the user again. I've changed it so that the user connect's instanly

**Comment by jigaryadav** [comment]
> Android seems to have race conditions while unmounting. Would be great to have a callback on disconnect handler at least. @slycoder do you have any ideas why does this work exactly like this?
> 
> Below I attach the code example of my temporary workaround:
> 
> ```js
> class TwilioVideoComponent extends Component {
>   state = {
>     readyToConnect: true
>   }
> 
>   componentWillUnmount () {
>     this.disconnect()
>   }
> 
>   connect (data) {
>     this.twilioRef.connect(data)
>   }
> 
>   reconnect (data) {
>     if (!this.state.readyToConnect) return
> 
>     this.disconnect()
> 
>     this.setState({ readyToConnect: false })
> 
>     setTimeout(() => {
>       if (!this.__mounted) return
> 
>       this.setState({ readyToConnect: true }, () => {
>         this.connect(data)
>       })
>     }, 3000)
>   }
> 
>   disconnect () {
>     this.twilioRef.disconnect()
>   }
> 
>   handleTwilioRef = (ref) => {
>     this.twilioRef = ref
>   }
> 
>   render () {
>     const { readyToConnect } = this.state
> 
>     if (!readyToConnect) {
>       return null
>     }
> 
>     return <TwilioVideoRN {...this.props.props} ref={this.handleTwilioRef} />
>   }
> }
> ```

this actually worked. let me know if we have any actual solution on this

**Comment by Balasnest** [comment]
I am facing similar issues. It's happening to set of users where this black screen arise. Tried with other set of users and its working properly.
What would be the issue? How to debug further?

**Comment by ihantanaHairun** [comment]
I try set Timeout 10 and it work :)
